fluxion/decoders/gstreamer.py

```diff
@@ -47,6 +50,20 @@ def decode(self, input_filepath: str, output_filepath: str):
                        stdout=subprocess.DEVNULL, check=True)
         return file_checksum(output_filepath)
 
+    @lru_cache
+    def check_run(self):
+        # pylint: disable=broad-except
+        elements = self.decoder_bin.split('!')
+        gst_inspect = f'gst-inspect-{self.gst_api}'
+        for element in elements:
+            element = element.strip().split()[0]
```

Comment:
This requires running several process, you can instead use this pipeline to check:
gst-launch-1.0 appsrc num-buffers=0 ! {decoder_bin} ! fakesink

Reply:
Yep, that is a good point. I will change it